src/mmw/apps/geoprocessing_api/views.py

```diff
@@ -1051,17 +1049,67 @@ def start_modeling_worksheet(request, format=None):
     of dictionaries, which should be posted to the /export/worksheet/ endpoint
     to get the actual Excel files.
     """
+    user = request.user if request.user.is_authenticated else None
     area_of_interest, _ = _parse_input(request)
 
-    pairs = huc12s_with_aois(area_of_interest)
+    matches = huc12s_with_aois(area_of_interest)
+    filenames = [{
+        'filename': '{}__{}'.format(m['id'], m['name'].replace(' ', '_')),
+        'name': m['name'],
+        'wkaoi': m['wkaoi'],
+        'aoi': m['aoi'],
+    } for m in matches]
+    huc12_ids = [int(m['wkaoi'][7:]) for m in matches]
```

Comment:
It seems like `huc12` has the same `btree` index as the database id does:

```
mmw=# \d boundary_huc12
              Table "public.boundary_huc12"
    Column     |            Type             | Modifiers
---------------+-----------------------------+-----------
 id            | integer                     | not null
 huc12         | character varying(12)       |
 name          | character varying(120)      |
 hutype        | character varying(254)      |
 geom          | geometry(MultiPolygon,4326) |
 geom_detailed | geometry(MultiPolygon,4326) |
Indexes:
    "boundary_huc12_pkey" PRIMARY KEY, btree (id)
    "huc12_idx" UNIQUE, btree (huc12)
    "boundary_huc12_geom_gist" gist (geom)
```

so it should be fine to use the `huc12` field. Didn't think of that originally because the `huc12` field was a later addition. Good catch!

Reply:
This is done in the most recent force pushed version.